### ISSUE TITLE
SA1300 - False positive on local methods

### ISSUE THREAD

**Issue by Mek7** [DotNetAnalyzers/StyleCopAnalyzers#3608]
For example:
```
public void First()
{
   void second() // SA1300 is fired here
   {
       // some code...
   }
}
```
There is false positive on SA1300 - Element should begin with upper-case letter - it is fired for local methods. But their scope is local only, so we think they should start with lower-case letter. So far, we are adding suppressions everywhere in our code, but this should probably be fixed in StyleCop.

**Comment by bjornhellander** [comment]
The style enforced by StyleCop Analyzers does not distinguish between methods with different visibility. This is intentional.
See https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1300.md

**Comment by Mek7** [comment]
Yes, I understand, but I suppose the rule was created when local methods were not possible. Now that they are possible in C#, shouldn't the rule be updated?

**Comment by Mek7** [comment]
I see. Thank you.

**Comment by sharwell** [comment]
PascalCase is the default, but there is an option to force it to prefer camelCase instead.
